### ISSUE TITLE
accessibility issue with inherited LJ subject icons

### ISSUE THREAD

**Issue by kareila** [dreamwidth/dw-free#1798]
http://www.dreamwidth.org/support/see_request?id=33421

We've received a report from a user having "neurological issues" caused by the rapid flashing of the side-to-side pair of eyes (htdocs/img/talk/sm10_eyes.gif), which are located in the optional subject icons on the comment form.

The user declined to bring up the issue in dw_suggestions or dw_accessibility because they don't have an account on the site, so I'm opening an issue here for discussion.

The two solutions that occur to me are to remove the animation entirely, or to slow the rate of refresh to something that isn't triggering.  Thoughts?

**Comment by zorkian** [comment]
Aren't those icons not visible by default?
 
--
Mark Smith
mark@qq.is
 
 
On Wed, May 25, 2016, at 09:21 PM, Jen wrote:
> http://www.dreamwidth.org/support/see_request?id=33421
> We've received a report from a user having "neurological issues"
> caused by the rapid flashing of the side-to-side pair of eyes
> (htdocs/img/talk/sm10_eyes.gif), which are located in the optional
> subject icons on the comment form.
> The user declined to bring up the issue in dw_suggestions or
> dw_accessibility because they don't have an account on the site, so
> I'm opening an issue here for discussion.
> The two solutions that occur to me are to remove the animation
> entirely, or to slow the rate of refresh to something that isn't
> triggering.  Thoughts?
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub[1]
>
 

Links:

  1. https://github.com/dreamwidth/dw-free/issues/1798

**Comment by kareila** [comment]
On the form, they are hidden by default, but if someone uses it, it is visible to anyone who views that comment.

**Comment by zorkian** [comment]
Got it. Seems reasonable to delete it/tone it down, if someone wants
to do that.
 
--
Mark Smith
mark@qq.is
 
 
On Wed, May 25, 2016, at 10:17 PM, Jen wrote:
> On the form, they are hidden by default, but if someone uses it, it is
> visible to anyone who views that comment.
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub[1]
>
 

Links:

  1. https://github.com/dreamwidth/dw-free/issues/1798#issuecomment-221779915

**Comment by rahaeli** [comment]
Here is the image slowed down 8x -- the delay changed from .1 second to .8 second. (I played around with the timing a bit and that's the one that looked best to me.)

![sm10_eyes](https://cloud.githubusercontent.com/assets/1716035/15591701/9ae07efc-236c-11e6-983c-cfc079d15a71.gif)

**Comment by kareila** [comment]
Looks good to me too.  Thanks for taking care of it!

Do you also want to submit the PR for replacing the existing image?  If not, I'll flag it for the beginner-bait list.

**Comment by rahaeli** [comment]
Nah, my repo is a mess and super out of date right now and I'd have to do a ton of work to get it PR-able -- anybody can just grab this.

**Comment by pinterface** [comment]
I kind of wonder if maybe the user who filed the complaint would be better off setting their browser to not animate gifs, or at least to only animate them once (e.g., by setting image.animation_mode in Firefox), but I've got ten minutes so why not?  Claim.

**Comment by rahaeli** [comment]
Yeah, that's a possible solution for any individual person, but the image did violate WCAG guidelines (no more than three flashes per second) and still should be changed on our end so we aren't triggering things for people that don't know those options exist.

**Comment by pinterface** [comment]
Instead of just increasing the delay between frames, another option would be to add in some in-between frames, to produce a smoother animation.  Something more like this:

![sm10_eyes_inbetween](https://cloud.githubusercontent.com/assets/2219765/15656239/aeb3f2fa-266a-11e6-8ff8-21a6b5f97fa9.gif)

That could use a little tweaking yet, but you get the idea.

**Comment by rahaeli** [comment]
I like that too!

**Comment by kareila** [comment]
I'm also good with that, if you want to go in that direction.

**Comment by pinterface** [comment]
Okay, tweaked it a little more.  100ms per inbetween frame, 200ms at each end, for a total of 500ms from one side to the other.  There's no science or strong intuition behind those numbers (I'm a programmer, not an artist!), so if you want timing adjustments, just let me know.  If it's good to go, I can update the pull request.

![sm10_eyes_inbetween-2b](https://cloud.githubusercontent.com/assets/2219765/15685856/4ec718c6-2733-11e6-9dfc-2a3c00e7be6b.gif)

**Comment by rahaeli** [comment]
I think that looks great, is much smoother, and doesn't flash anymore. A++, would thumbs up again.
